templates/definition/meter/sma-hybrid.yaml

```diff
@@ -17,50 +18,144 @@ params:
   {{- end }}
   {{- if eq .usage "battery" }}
+  power:
     source: calc
-    mul:
+    add:
       - source: modbus
         {{- include "modbus" . | indent 6 }}
-        register: # manual non-sunspec register configuration
-          address: 30851 # SMA Modbus Profile: Battery voltage
-          type: holding
+        register:
+          address: 31395 # SMA Modbus Profile: BatDsch.CurBatDsch
+          type: input
           decode: uint32nan
-        scale: 0.01
       - source: modbus
         {{- include "modbus" . | indent 6 }}
-        register: # manual non-sunspec register configuration
-          address: 30843 # SMA Modbus Profile: Battery current
-          type: holding
-          decode: int32nan
-        scale: 0.001
+        register:
+          address: 31393 # SMA Modbus Profile: BatChrg.CurBatCha
+          type: input
+          decode: uint32nan
+        scale: -1
+  energy:
+    source: modbus
+    {{- include "modbus" . | indent 2 }}
+    register:
+      address: 31401 # SMA Modbus Profile: CmpBMS.GetBatDschWh
+      type: holding
+      decode: uint64nan
+    scale: 0.001
   soc:
     source: modbus
     {{- include "modbus" . | indent 2 }}
-    register: # manual non-sunspec register configuration
-      address: 30845 # SMA Modbus Profile: Battery state of charge
+    register:
+      address: 30845 # SMA Modbus Profile: Bat.ChaStt
       type: holding
       decode: uint32nan
+  batterymode:
+    source: watchdog
+    timeout: {{ .watchdog }} # re-write at timeout/2
+    reset: 1 # reset watchdog on normal
+    set:
+      source: switch
+      switch:
+      - case: 1 # normal
+        set:
+          source: const
+          value: 10000 # Maximale Wirkleistung
+          set:
+            source: modbus
+            {{- include "modbus" . | indent 10 }}
+            timeout: {{ .timeout }}
+            register:
+              address: 44039 # SMA Modbus Profile: Inverter.WModCfg.WCtlComCfg.WSptMaxNom
+              type: writemultiple
+              decode: int32
+      - case: 2 # hold
+        set:
+          source: const
+          value: 0 # Maximale Wirkleistung
+          set:
+            source: modbus
+            {{- include "modbus" . | indent 10 }}
+            timeout: {{ .timeout }}
+            register:
+              address: 44039 # SMA Modbus Profile: Inverter.WModCfg.WCtlComCfg.WSptMaxNom
+              type: writemultiple
+              decode: int32
+      - case: 3 # charge
+        set:
+          source: const
+          value: -10000 # Maximale Wirkleistung
+          set:
+            source: modbus
+            {{- include "modbus" . | indent 10 }}
+            timeout: {{ .timeout }}
+            register:
+              address: 44039 # SMA Modbus Profile: Inverter.WModCfg.WCtlComCfg.WSptMaxNom
+              type: writemultiple
+              decode: int32
   {{- if .capacity }}
   capacity: {{ .capacity }} # kWh
   {{- end }}
   {{- end }}
+  {{- if eq .usage "grid" }}
```

Comment:
`grid` ist keine gültige Wahl für das Template (s.o.). Falls das geht wäre es schön grid nach oben zu schieben wie in den anderen Templates auch.

Reply:
...womit sich die Frage stellt ob wir das brauchen bzw. wo `grid` bisher her kam?

Reply:
Ich hatte das nur mal drin gelassen weil es so mühevoll war es rauszusuchen.
grid kommt bei SMA ja meist vom SHM.
Man kann dessen Messwerte aber eben auch über den WR per Modbus abfragen...